src/shared/model/Errors.ts

```diff
@@ -88,3 +88,31 @@ export enum UnsubscribeErrors {
 export enum SubscribeErrors {
 	GENERIC = 'There was a problem subscribing, please try again.',
 }
+
+export interface StructuredGatewayError {
+	message: string;
+	severity: 'BAU' | 'UNEXPECTED';
+}
+
+export type GatewayError = StructuredGatewayError | string;
+
+// ensures we have a structured error, if the original was a string it applies default values for the rest of the fields
+export function asStructuredError(
+	error?: GatewayError,
+): StructuredGatewayError | undefined {
+	if (typeof error === 'string') {
+		return {
+			message: error,
+			severity: 'UNEXPECTED',
+		};
+	}
+	return error;
+}
+
+// ensures we get a string error message, if the original was a structured error it throws away the rest of the fields
+export function extractMessage(error?: GatewayError): string | undefined {
+	if (typeof error === 'object') {
+		return error.message;
+	}
+	return error;
+}
```

Comment:
I think it would be easier to check this the other way around, i.e check for a sting first, if it's not a string then it's an object and return the message then.

```suggestion
export function extractMessage(error?: GatewayError): string | undefined {
	if (typeof error === 'string') {
		return error;
	}
	return error.message;
}
```

Reply:
> I think it would be easier to check this the other way around, i.e check for a sting first, if it's not a string then it's an object and return the message then.
I'm not sure why it's easier to check for string but the only reason I did it this way was because otherwise I have to account for undefined when attempting to return  `error.message`.  I guess I can return `error?.message` instead

Reply:
yeah you'd have to return `error?.message` as `error` might be undefined.